app/models/course/assessment/question/programming.rb

```diff
@@ -272,6 +273,23 @@ def validate_codaveri_question
                  'Activate it in the course setting or switch this question into a non-codaveri type.')
     end
   end
+
+  def validate_koditsu_question_in_koditsu_assessment
+    return unless koditsu_enabled?
+
+    unless KoditsuAsyncApiService.language_valid_for_koditsu?(language)
+      errors.add(:base, 'Language type is not compatible with Koditsu')
+    end
+
+    true
+  end
+
+  def koditsu_enabled?
+    question_assessment = acting_as.question_assessments&.first
+    assessment = question_assessment&.assessment
+
+    assessment&.course&.component_enabled?(Course::KoditsuPlatformComponent) && assessment&.is_koditsu_enabled
+  end
```

Comment:
What is this validation for? Is it for duplication?  
By right, our schema is structured such that questions do not know which assessments they are attached to. Which means Koditsu validity should come from assessment level.  
Duplication behaviour requires spec-ing, e.g. do we allow duplicating a C++ programming qns from non-Koditsu assessment to Koditsu-enabled assessment, and does an associated Koditsu question get created alongside?

Reply:
this is for creating the question in general
will be useful for, let's say, if I try creating the Java question in non-koditsu assessment, but in the middle of me creating it, koditsu got enabled

then, this validation will reject such question to even be created in the first place

Reply:
This validation likely belongs in `Course::QuestionAssessment` instead of here. Responsibility not in question itself. Which is also why the spec file had to awkwardly introduce the `[qa]` relationship.